x-pack/plugins/lists/server/services/exception_lists/utils.ts

```diff
@@ -316,61 +304,74 @@ export const transformUpdateCommentsToComments = ({
+  if (comments == null && existingComments.length > 0) {
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
     throw new ErrorWithStatusCode(
-      'Comments cannot be deleted, only new comments may be added',
-      403
+      'Empty "[]" passed to "comments" - detected existing comments. Comment cannot be deleted.',
+      401
     );
-  } else {
-    return newComments.flatMap((c, index) => {
-      const existingComment = existingComments[index];
+  }
 
-      if (commentsSchema.is(existingComment) && !commentsSchema.is(c)) {
-        throw new ErrorWithStatusCode(
-          'When trying to update a comment, "created_at" and "created_by" must be present',
-          403
-        );
-      } else if (existingComment == null && commentsSchema.is(c)) {
-        throw new ErrorWithStatusCode('Only new comments may be added', 403);
-      } else if (
-        commentsSchema.is(c) &&
-        existingComment != null &&
-        isCommentEqual(c, existingComment)
-      ) {
-        return existingComment;
-      } else if (commentsSchema.is(c) && existingComment != null) {
-        return transformUpdateComments({ comment: c, existingComment, user });
-      } else {
-        return transformCreateCommentsToComments({ comments: [c], user }) ?? [];
-      }
-    });
+  if (comments != null && comments.length < existingComments.length) {
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
+    throw new ErrorWithStatusCode('Comment cannot be deleted.', 401);
   }
+
+  if (comments == null && existingComments.length === 0) {
+    return [];
+  }
+
+  const incomingComments = comments ?? [];
+
+  return incomingComments.map<Comment>((item, index) => {
+    const matchingCommentByIndex = existingComments[index];
+    const idMatches = matchingCommentByIndex && item.id === matchingCommentByIndex.id;
+
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
+    if (!idMatches && index < existingComments.length) {
+      throw new ErrorWithStatusCode(`Cannot delete comment id: ${matchingCommentByIndex.id}`, 401);
+    } else if (!matchingCommentByIndex && index > existingComments.length - 1) {
+      const [newComment] = transformCreateCommentsToComments({
+        incomingComments: [{ ...item }],
+        user,
+      });
+      return newComment;
+    } else {
+      return transformUpdateComments({
+        existingComment: matchingCommentByIndex,
+        incomingComment: item,
+        user,
+      });
+    }
+  });
 };
 
 export const transformUpdateComments = ({
-  comment,
+  incomingComment,
   existingComment,
   user,
 }: {
-  comment: Comments;
-  existingComment: Comments;
+  incomingComment: UpdateComment;
+  existingComment: Comment;
   user: string;
-}): Comments => {
-  if (comment.created_by !== user) {
+}): Comment => {
+  if (incomingComment.id == null) {
```

Comment:
I'm seeing some issues when trying to edit an exception that has been changed at the same time by the same user (in another browser) or by another user. The steps I'm following are:

1. Create an exception with one comment in one browser (Firefox)
2. In another browser (Chrome) click "Edit" on that recently added exception but **don't edit it and don't click submit**.
3. Back in the Firefox window edit the exception by adding a new comment
4. In the Chrome window add a new comment and clicking submit

Before this change, in the master branch, I get a `403` on `PUT /api/exception_lists/items` error with the following message:
```
When trying to update a comment, "created_at" and "created_by" must be present
```
After this change, in this branch, I get a `401 unathorized` error on `PUT /api/exception_lists/items` and get **logged out of Kibana**.
![image](https://user-images.githubusercontent.com/616158/88353568-0129e780-cd2c-11ea-8eeb-723cb0bd0d84.png)

Reply:
Thanks so much for catching this Pedro. I updated the API to currently be append only. I was doing validations and transformations prior to the version check. I removed much of this and am now just checking that the incoming comments are append only. Still working to have full CRUD in an upcoming release.